lib/shopify_transporter/record_builder/product_record_builder.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+#
+module ShopifyTransporter
+  class ProductRecordBuilder < RecordBuilder
+    def initialize(key_name, key_required)
+      super(key_name, key_required)
+    end
+
+    def build(input)
+      validate_related(input)
+
+      if has_parent?(input)
+        yield parent_record(input)
+        return
+      end
+
+      yield record_from(input)
+    end
+
+    private
+
+    def has_parent?(product)
+      product['parent_id'].present?
+    end
+
+    def parent_record(product)
+      parent_id = product['parent_id']
+      record = @instances[parent_id] ||= {}
+      @last_record = record
```

Comment:
what is this used for again?

Reply:
This is used if the row does not have a key or a parent_id. The record builder returns the last processed object.

This supports data like this:
```
Name, Line Item 1
ORD1, Banana
, Apple
```

(It will tie Apple to the last processed Order, ORD1)